packages/phisherman/src/lib/Phisherman.ts

```diff
@@ -0,0 +1,90 @@
+import fetch, { Headers } from 'node-fetch';
+import { fromAsync, isErr } from '@sapphire/framework';
+
+export class Phisherman {
+	/**
+	 * The options for phisherman
+	 */
+	public options;
+	public constructor(options: PhishermanOptions | undefined) {
+		this.options = options;
+		void this.checkApiKey(this.options!.apiKey).then((res) => {
+			if (res) throw new Error('Invalid API key');
+		});
```

Comment:
Don't use promises in constructor. That doesn't really work. Instead dedicate this to a method called `init`